src/HMRC.ESFA.Levy.Api.Client/ApprenticeshipLevyApiClient.cs

```diff
@@ -15,16 +16,18 @@ namespace HMRC.ESFA.Levy.Api.Client
     public class ApprenticeshipLevyApiClient : IApprenticeshipLevyApiClient
     {
         private readonly HttpClient _client;
-
+        private readonly IPaymentStatusProcessor _paymentStatusProcessor;
         private const string DateFormat = "yyyy-MM-dd";
 
         /// <summary>
         /// Default constuctor
         /// </summary>
         /// <param name="client">A configured HttpClient, alternatively use ApprenticeshipLevyApiClient.CreateHttpClient(token, url)</param>
-        public ApprenticeshipLevyApiClient(HttpClient client)
+        /// <param name="paymentStatusProcessor"></param>
+        public ApprenticeshipLevyApiClient(HttpClient client, IPaymentStatusProcessor paymentStatusProcessor)
```

Comment:
We should keep the public cstr

```c#
public ApprenticeshipLevyApiClient(HttpClient client)
```

and you can use an internal cstr

```c#
internal ApprenticeshipLevyApiClient(HttpClient client, IPaymentStatusProcessor paymentStatusProcessor)
 : this(client, new PaymentStatusProcessor()){
}
```

Reply:
but.....

http://chapterbreak.net/wp-content/uploads/2015/12/only-one.gif

Reply:
I'm kind of 50/50 on doing either this, or just concreting up the processor inside the function that uses it, which is equally smelly but a little more honest/direct.  Thoughts?